dissect/util/compression/lzvn.py

```diff
@@ -0,0 +1,237 @@
+            # the first byte. We first extract the literal length so we know how long
+            # the opcode is, then check that the source can hold both this opcode and
+            # at least one byte of the next (because any valid input stream must be
+            # terminated with an eos token).
+            opc_len = 2
+            L = _extract(opc, 8, 6, 2)
+            M = _extract(opc, 8, 3, 3) + 3
+
+            if src_size <= opc_len + L:
+                break
+
+            D = _extract(opc, 8, 0, 3) << 8 | src.read(1)[0]
+
+        elif opc in OP_MED_D:
+            # "medium distance": This is a minor variant of the "small distance"
+            # encoding, where we will now use two extra bytes instead of one to encode
+            # the restof the match length and distance. This allows an extra two bits
+            # for the match length, and an extra three bits for the match distance. The
+            # full structure of the opcode is 101LLMMM DDDDDDMM DDDDDDDD LITERAL.
+            opc_len = 3
+            L = _extract(opc, 8, 3, 2)
+
+            if src_size <= opc_len + L:
+                break
+
+            (opc23,) = _H.unpack(src.read(2))
+            M = (_extract(opc, 8, 0, 3) << 2 | _extract(opc23, 16, 0, 2)) + 3
+            D = _extract(opc23, 16, 2, 14)
+
+        elif opc in OP_LRG_D:
+            # "large distance": This is another variant of the "small distance"
+            # encoding, where we will now use two extra bytes to encode the match
+            # distance, which allows distances up to 65535 to be represented. The full
+            # structure of the opcode is LLMMM111 DDDDDDDD DDDDDDDD LITERAL.
+            opc_len = 3
+            L = _extract(opc, 8, 6, 2)
+            M = _extract(opc, 8, 3, 3) + 3
+
+            if src_size <= opc_len + L:
+                break
+
+            (D,) = _H.unpack(src.read(2))
+
+        elif opc in OP_PRE_D:
+            # "previous distance": This opcode has the structure LLMMM110, where the
+            # length of the literal (0-3 bytes) is encoded by the high 2 bits of the
+            # first byte. We first extract the literal length so we know how long
+            # the opcode is, then check that the source can hold both this opcode and
+            # at least one byte of the next (because any valid input stream must be
+            # terminated with an eos token).
+            opc_len = 1
+            L = _extract(opc, 8, 6, 2)
+            M = _extract(opc, 8, 3, 3) + 3
+
+            if src_size <= opc_len + L:
+                break
+
+        elif opc in OP_SML_M:
+            # "small match": This opcode has no literal, and uses the previous match
+            # distance (i.e. it encodes only the match length), in a single byte as
+            # 1111MMMM.
+            opc_len = 1
+            L = None
+            M = _extract(opc, 8, 0, 4)
+
+            if src_size <= opc_len:
+                break
+
```

Comment:
The reference puts the size check before the extract here. Maybe it matters.
```suggestion
            opc_len = 1
            if src_size <= opc_len:
                break

            M = _extract(opc, 8, 0, 4)
            L = None
```

Reply:
I saw this one but I don't think it matters. I deliberately changed it to be more consistent.